src/measure_theory/measure_space.lean

```diff
@@ -1735,6 +1750,77 @@ end
 
 end measure_sub
 
+
+lemma restrict_sub_eq_restrict_sub_restrict {Ω : Type*} [measurable_space Ω]
+  (μ ν : measure Ω) {s : set Ω} (h_meas_s : is_measurable s) :
+  (μ - ν).restrict s = (μ.restrict s) - (ν.restrict s) :=
+begin
+  repeat {rw sub_def},
+  have h_nonempty : {d | μ ≤ d + ν}.nonempty,
+  { apply @set.nonempty_of_mem _ _ μ, simp, intros t h_meas,
+    apply le_add_right (le_refl (μ t)) },
+  rw restrict_Inf_eq_Inf_restrict h_nonempty h_meas_s,
+  have h_Inf_le_Inf : ∀ s' t' : set (measure Ω),
+                      (∀ b ∈ t', ∃ a ∈ s', a ≤ b) → Inf s' ≤ Inf t',
+  { intros s' t' h, 
+    rw le_Inf_iff, intros b h_b_in_t', 
+    have h_exists_a := h b h_b_in_t', cases h_exists_a with a h_a, 
+    cases h_a with h_a_in_s' h_a_le_b,
+    apply Inf_le_of_le h_a_in_s' h_a_le_b },  
+  apply le_antisymm,
+  { apply h_Inf_le_Inf,
+    intros ν' h_ν'_in, simp at h_ν'_in, apply exists.intro (ν'.restrict s),
+    split,
+    { simp, apply exists.intro (ν' + (⊤ : measure_theory.measure Ω).restrict sᶜ),
+      split,
+      { rw [add_assoc, add_comm _ ν, ← add_assoc, measure_theory.measure.le_iff],
+        intros t h_meas_t,
+        have h_inter_inter_eq_inter : ∀ t' : set Ω , t ∩ t' ∩ t' = t ∩ t',
+        { intro t', rw set.inter_eq_self_of_subset_left, apply set.inter_subset_right t t' },
+        have h_meas_t_inter_s : is_measurable (t ∩ s) :=
+        is_measurable.inter h_meas_t h_meas_s,
+        rw [← inter_add_inter_compl μ t s h_meas_t h_meas_s,
+            ← inter_add_inter_compl 
+              (ν' + ν + (⊤ : measure Ω).restrict sᶜ) t s h_meas_t h_meas_s],
+        apply add_le_add _ _; rw add_apply,
+        { have h_restrict : ∀ μ₂ : measure Ω, μ₂ (t ∩ s) = μ₂.restrict s (t ∩ s),
+          { intro μ₂, rw [restrict_apply h_meas_t_inter_s], 
+            rw [(h_inter_inter_eq_inter s)] },
+          apply le_add_right _,
+          rw [add_apply, h_restrict μ, h_restrict ν], apply h_ν'_in _ h_meas_t_inter_s },
+        cases (@set.eq_empty_or_nonempty _ (t ∩ sᶜ)) with h_inter_empty h_inter_nonempty,
+        { simp [h_inter_empty] },
+        { have h_meas_inter_compl := 
+          is_measurable.inter h_meas_t (is_measurable.compl h_meas_s),
```

Comment:
If you want, you can shorten things like this to the following (also in some other places in your proof):
```suggestion
          h_meas_t.inter h_meas_s.compl,
```